From 1e03c11dff31b2556a1ce5c849fc4f165c3a3461 Mon Sep 17 00:00:00 2001
From: Ray Rashif <schivmeister@gmail.com>
Date: Thu, 22 Oct 2009 22:38:00 +0800
Subject: [PATCH] Fix JOST 0.4.6 building with GCC 4.4

This is just added on to the previous gcc43.patch, which until today I'm
still amazed I actually wrote an entire script to produce. Well, 0.5.4
actually looks totally  _broken_ with 4.4, hence coming back to the
trash to dig this out. Lucky it's just 5 files this time, and not 50.

Signed-off-by: Ray Rashif <schivmeister@gmail.com>
---
 .../juce_linux_AudioCDReader.cpp                   |    1 +
 .../platform_specific_code/juce_linux_Files.cpp    |    1 +
 .../platform_specific_code/juce_linux_Network.cpp  |    1 +
 .../juce_linux_SystemStats.cpp                     |    2 ++
 .../juce_linux_Windowing.cpp                       |   18 ++++++++++--------
 .../application/juce_ApplicationCommandManager.cpp |    1 +
 .../application/juce_DeletedAtShutdown.cpp         |    1 +
 .../audio_file_formats/juce_AiffAudioFormat.cpp    |    1 +
 .../juce_AudioSubsectionReader.cpp                 |    1 +
 .../audio_file_formats/juce_WavAudioFormat.cpp     |    1 +
 .../audio/audio_sources/juce_AudioSourcePlayer.cpp |    1 +
 .../audio_sources/juce_BufferingAudioSource.cpp    |    1 +
 .../audio/audio_sources/juce_MixerAudioSource.cpp  |    1 +
 .../juce_PositionableMixerAudioSource.cpp          |    1 +
 .../juce_PositionableResamplingAudioSource.cpp     |    1 +
 .../audio_sources/juce_ResamplingAudioSource.cpp   |    1 +
 .../audio_sources/juce_SequenceAudioSource.cpp     |    1 +
 .../audio/devices/juce_AudioDeviceManager.cpp      |    1 +
 .../audio/dsp/juce_AudioSampleBuffer.cpp           |    1 +
 .../juce_appframework/audio/dsp/juce_IIRFilter.cpp |    1 +
 .../audio/midi/juce_MidiBuffer.cpp                 |    1 +
 .../audio/midi/juce_MidiKeyboardState.cpp          |    1 +
 .../audio/midi/juce_MidiMessage.cpp                |    1 +
 .../audio/midi/juce_MidiMessageSequence.cpp        |    1 +
 .../audio/plugins/juce_KnownPluginList.cpp         |    1 +
 .../audio/plugins/juce_PluginDirectoryScanner.cpp  |    1 +
 .../audio/plugins/juce_PluginListComponent.cpp     |    1 +
 .../audio/processors/juce_AudioProcessor.cpp       |    1 +
 .../audio/processors/juce_AudioProcessorGraph.cpp  |    1 +
 .../audio/processors/juce_AudioProcessorPlayer.cpp |    1 +
 .../audio/synthesisers/juce_Synthesiser.cpp        |    1 +
 .../documents/juce_UndoManager.cpp                 |    1 +
 .../events/juce_ActionListenerList.cpp             |    1 +
 .../events/juce_ChangeListenerList.cpp             |    1 +
 .../events/juce_MessageListener.cpp                |    1 +
 .../gui/components/buttons/juce_Button.cpp         |    1 +
 .../gui/components/controls/juce_ComboBox.cpp      |    1 +
 .../gui/components/controls/juce_Label.cpp         |    1 +
 .../gui/components/controls/juce_ListBox.cpp       |    1 +
 .../gui/components/controls/juce_Slider.cpp        |    1 +
 .../controls/juce_TableHeaderComponent.cpp         |    1 +
 .../gui/components/controls/juce_TableListBox.cpp  |    1 +
 .../gui/components/controls/juce_TextEditor.cpp    |    1 +
 .../gui/components/controls/juce_Toolbar.cpp       |    1 +
 .../gui/components/controls/juce_TreeView.cpp      |    1 +
 .../juce_DirectoryContentsDisplayComponent.cpp     |    1 +
 .../filebrowser/juce_DirectoryContentsList.cpp     |    1 +
 .../filebrowser/juce_FileBrowserComponent.cpp      |    1 +
 .../filebrowser/juce_FilenameComponent.cpp         |    1 +
 .../gui/components/juce_Component.cpp              |    1 +
 .../gui/components/juce_Desktop.cpp                |    1 +
 .../keyboard/juce_KeyMappingEditorComponent.cpp    |    1 +
 .../keyboard/juce_KeyPressMappingSet.cpp           |    1 +
 .../keyboard/juce_KeyboardFocusTraverser.cpp       |    1 +
 .../components/layout/juce_ComponentAnimator.cpp   |    1 +
 .../components/layout/juce_MultiDocumentPanel.cpp  |    1 +
 .../gui/components/layout/juce_ScrollBar.cpp       |    1 +
 .../layout/juce_StretchableLayoutManager.cpp       |    1 +
 .../gui/components/layout/juce_TabbedButtonBar.cpp |    1 +
 .../gui/components/layout/juce_TabbedComponent.cpp |    1 +
 .../gui/components/menus/juce_MenuBarModel.cpp     |    1 +
 .../gui/components/menus/juce_PopupMenu.cpp        |    1 +
 .../gui/components/mouse/juce_MouseCursor.cpp      |    1 +
 .../gui/components/special/juce_ColourSelector.cpp |    1 +
 .../special/juce_MidiKeyboardComponent.cpp         |    1 +
 .../components/special/juce_OpenGLComponent.cpp    |    1 +
 .../gui/components/windows/juce_AlertWindow.cpp    |    1 +
 .../gui/components/windows/juce_ComponentPeer.cpp  |    1 +
 .../gui/components/windows/juce_DocumentWindow.cpp |    1 +
 .../gui/components/windows/juce_TopLevelWindow.cpp |    1 +
 .../gui/graphics/brushes/juce_GradientBrush.cpp    |    1 +
 .../gui/graphics/colour/juce_ColourGradient.cpp    |    1 +
 .../gui/graphics/contexts/juce_EdgeTable.cpp       |    1 +
 .../gui/graphics/contexts/juce_Graphics.cpp        |    1 +
 .../juce_LowLevelGraphicsPostScriptRenderer.cpp    |    2 ++
 .../juce_LowLevelGraphicsSoftwareRenderer.cpp      |    1 +
 .../graphics/drawables/juce_DrawableComposite.cpp  |    1 +
 .../gui/graphics/drawables/juce_SVGParser.cpp      |    1 +
 .../gui/graphics/fonts/juce_GlyphArrangement.cpp   |    1 +
 .../gui/graphics/fonts/juce_Typeface.cpp           |    1 +
 .../gui/graphics/geometry/juce_Path.cpp            |    1 +
 .../gui/graphics/geometry/juce_RectangleList.cpp   |    1 +
 .../imaging/image_file_formats/juce_GIFLoader.cpp  |    1 +
 .../imaging/image_file_formats/juce_JPEGLoader.cpp |    1 +
 .../gui/graphics/imaging/juce_Image.cpp            |    1 +
 .../gui/graphics/imaging/juce_ImageCache.cpp       |    1 +
 juce/src/juce_core/containers/juce_BitArray.cpp    |    1 +
 juce/src/juce_core/containers/juce_MemoryBlock.cpp |    1 +
 juce/src/juce_core/cryptography/juce_BlowFish.cpp  |    1 +
 juce/src/juce_core/cryptography/juce_MD5.cpp       |    1 +
 .../juce_core/io/files/juce_FileOutputStream.cpp   |    1 +
 juce/src/juce_core/io/juce_OutputStream.cpp        |    1 +
 juce/src/juce_core/io/network/juce_Socket.cpp      |    1 +
 .../io/streams/juce_BufferedInputStream.cpp        |    1 +
 .../io/streams/juce_MemoryInputStream.cpp          |    1 +
 juce/src/juce_core/misc/juce_Uuid.cpp              |    1 +
 juce/src/juce_core/misc/juce_ZipFile.cpp           |    1 +
 .../src/juce_core/text/juce_CharacterFunctions.cpp |    2 ++
 juce/src/juce_core/text/juce_String.cpp            |    1 +
 juce/src/juce_core/text/juce_StringArray.cpp       |    1 +
 juce/src/juce_core/threads/juce_Thread.cpp         |    1 +
 juce/src/juce_core/threads/juce_ThreadPool.cpp     |    1 +
 .../src/juce_core/threads/juce_TimeSliceThread.cpp |    1 +
 jucetice/src/dependancies/libsocket/UDPSocket.cpp  |    3 +++
 .../rubberband/HighFrequencyAudioCurve.cpp         |    2 ++
 .../rubberband/RubberBandStretcher.cpp             |    2 ++
 .../rubberband/SpectralDifferenceAudioCurve.cpp    |    2 ++
 .../dependancies/rubberband/StretchCalculator.cpp  |    2 ++
 .../rubberband/StretcherChannelData.cpp            |    2 ++
 .../src/dependancies/rubberband/StretcherImpl.cpp  |    3 +++
 .../dependancies/rubberband/StretcherProcess.cpp   |    3 +++
 jucetice/src/dependancies/rubberband/Thread.cpp    |    2 ++
 jucetice/src/jucetice_OpenSoundController.cpp      |    1 +
 113 files changed, 137 insertions(+), 8 deletions(-)

diff --git a/juce/build/linux/platform_specific_code/juce_linux_AudioCDReader.cpp b/juce/build/linux/platform_specific_code/juce_linux_AudioCDReader.cpp
index 7bcaa6f..67d9839 100755
--- a/juce/build/linux/platform_specific_code/juce_linux_AudioCDReader.cpp
+++ b/juce/build/linux/platform_specific_code/juce_linux_AudioCDReader.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../src/juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/build/linux/platform_specific_code/juce_linux_Files.cpp b/juce/build/linux/platform_specific_code/juce_linux_Files.cpp
index 2084a1f..0a0e74f 100755
--- a/juce/build/linux/platform_specific_code/juce_linux_Files.cpp
+++ b/juce/build/linux/platform_specific_code/juce_linux_Files.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstdio>
 #include "linuxincludes.h"
 #include "../../../src/juce_core/basics/juce_StandardHeader.h"
 
diff --git a/juce/build/linux/platform_specific_code/juce_linux_Network.cpp b/juce/build/linux/platform_specific_code/juce_linux_Network.cpp
index 75b3bc2..5d97faf 100755
--- a/juce/build/linux/platform_specific_code/juce_linux_Network.cpp
+++ b/juce/build/linux/platform_specific_code/juce_linux_Network.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "linuxincludes.h"
 #include <netdb.h>
 #include <arpa/inet.h>
diff --git a/juce/build/linux/platform_specific_code/juce_linux_SystemStats.cpp b/juce/build/linux/platform_specific_code/juce_linux_SystemStats.cpp
index bfa06be..cf32783 100755
--- a/juce/build/linux/platform_specific_code/juce_linux_SystemStats.cpp
+++ b/juce/build/linux/platform_specific_code/juce_linux_SystemStats.cpp
@@ -29,6 +29,8 @@
   ==============================================================================
 */
 
+#include <cstdio>
+#include <cstring>
 #include "linuxincludes.h"
 #include "../../../src/juce_core/basics/juce_StandardHeader.h"
 #include <sys/sysinfo.h>
diff --git a/juce/build/linux/platform_specific_code/juce_linux_Windowing.cpp b/juce/build/linux/platform_specific_code/juce_linux_Windowing.cpp
index 8b58ccd..96b1d7a 100755
--- a/juce/build/linux/platform_specific_code/juce_linux_Windowing.cpp
+++ b/juce/build/linux/platform_specific_code/juce_linux_Windowing.cpp
@@ -29,6 +29,8 @@
   ==============================================================================
 */
 
+#include <cstdio>
+#include <cstring>
 #include "../../../juce_Config.h"
 
 #if JUCE_BUILD_GUI_CLASSES
@@ -565,7 +567,7 @@ public:
     }
 
     ~XBitmapImage()
-    {
+    {
 #if JUCE_USE_XSHM
         if (usingXShm)
         {
@@ -584,7 +586,7 @@ public:
             xImage->data = 0;
             XDestroyImage (xImage);
         }
-
+
         if (! is16Bit)
             imageData = 0; // to stop the base class freeing this (for the 16-bit version we want it to free it)
     }
@@ -592,10 +594,10 @@ public:
     void blitToWindow (Window window, int dx, int dy, int dw, int dh, int sx, int sy)
     {
         static GC gc = 0;
-
+
         if (gc == 0)
             gc = DefaultGC (display, DefaultScreen (display));
-
+
         if (is16Bit)
         {
             const uint32 rMask = xImage->red_mask;
@@ -761,7 +763,7 @@ public:
             wy = y;
             ww = jmax (1, w);
             wh = jmax (1, h);
-
+
             if (! mapped)
             {
                 // Make sure the Window manager does what we want
@@ -780,7 +782,7 @@ public:
                                wy - windowBorder.getTop(),
                                ww + windowBorder.getLeftAndRight(),
                                wh + windowBorder.getTopAndBottom());
-
+
             if (! deletionChecker.hasBeenDeleted())
             {
                 updateBorderSize();
@@ -1035,7 +1037,7 @@ public:
                     False,
                     SubstructureRedirectMask | SubstructureNotifyMask,
                     &ev);
-
+
         XFlush (display);
 
         handleBroughtToFront();
@@ -1286,7 +1288,7 @@ public:
                 updateKeyModifiers (buttonPressEvent->state);
 
                 if (buttonMsg)
-                {
+                {
                     Window wRoot = 0, wParent = 0;
                     Window* wChild = 0;
                     unsigned int numChildren;
diff --git a/juce/src/juce_appframework/application/juce_ApplicationCommandManager.cpp b/juce/src/juce_appframework/application/juce_ApplicationCommandManager.cpp
index 9a785de..9d0fa78 100755
--- a/juce/src/juce_appframework/application/juce_ApplicationCommandManager.cpp
+++ b/juce/src/juce_appframework/application/juce_ApplicationCommandManager.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/application/juce_DeletedAtShutdown.cpp b/juce/src/juce_appframework/application/juce_DeletedAtShutdown.cpp
index 430a30e..ef4ec4d 100755
--- a/juce/src/juce_appframework/application/juce_DeletedAtShutdown.cpp
+++ b/juce/src/juce_appframework/application/juce_DeletedAtShutdown.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/audio_file_formats/juce_AiffAudioFormat.cpp b/juce/src/juce_appframework/audio/audio_file_formats/juce_AiffAudioFormat.cpp
index 563da3c..790c0f3 100755
--- a/juce/src/juce_appframework/audio/audio_file_formats/juce_AiffAudioFormat.cpp
+++ b/juce/src/juce_appframework/audio/audio_file_formats/juce_AiffAudioFormat.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/audio_file_formats/juce_AudioSubsectionReader.cpp b/juce/src/juce_appframework/audio/audio_file_formats/juce_AudioSubsectionReader.cpp
index 29aa4b3..99db807 100755
--- a/juce/src/juce_appframework/audio/audio_file_formats/juce_AudioSubsectionReader.cpp
+++ b/juce/src/juce_appframework/audio/audio_file_formats/juce_AudioSubsectionReader.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/audio_file_formats/juce_WavAudioFormat.cpp b/juce/src/juce_appframework/audio/audio_file_formats/juce_WavAudioFormat.cpp
index 3b7f17e..ae3c874 100755
--- a/juce/src/juce_appframework/audio/audio_file_formats/juce_WavAudioFormat.cpp
+++ b/juce/src/juce_appframework/audio/audio_file_formats/juce_WavAudioFormat.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/audio_sources/juce_AudioSourcePlayer.cpp b/juce/src/juce_appframework/audio/audio_sources/juce_AudioSourcePlayer.cpp
index b1f1b38..53296e3 100755
--- a/juce/src/juce_appframework/audio/audio_sources/juce_AudioSourcePlayer.cpp
+++ b/juce/src/juce_appframework/audio/audio_sources/juce_AudioSourcePlayer.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/audio_sources/juce_BufferingAudioSource.cpp b/juce/src/juce_appframework/audio/audio_sources/juce_BufferingAudioSource.cpp
index e4edd21..22bd644 100755
--- a/juce/src/juce_appframework/audio/audio_sources/juce_BufferingAudioSource.cpp
+++ b/juce/src/juce_appframework/audio/audio_sources/juce_BufferingAudioSource.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/audio_sources/juce_MixerAudioSource.cpp b/juce/src/juce_appframework/audio/audio_sources/juce_MixerAudioSource.cpp
index 574ca66..0e6923b 100755
--- a/juce/src/juce_appframework/audio/audio_sources/juce_MixerAudioSource.cpp
+++ b/juce/src/juce_appframework/audio/audio_sources/juce_MixerAudioSource.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/audio_sources/juce_PositionableMixerAudioSource.cpp b/juce/src/juce_appframework/audio/audio_sources/juce_PositionableMixerAudioSource.cpp
index d3ded6c..eb7de90 100755
--- a/juce/src/juce_appframework/audio/audio_sources/juce_PositionableMixerAudioSource.cpp
+++ b/juce/src/juce_appframework/audio/audio_sources/juce_PositionableMixerAudioSource.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/audio_sources/juce_PositionableResamplingAudioSource.cpp b/juce/src/juce_appframework/audio/audio_sources/juce_PositionableResamplingAudioSource.cpp
index 2540cdc..a81bb56 100755
--- a/juce/src/juce_appframework/audio/audio_sources/juce_PositionableResamplingAudioSource.cpp
+++ b/juce/src/juce_appframework/audio/audio_sources/juce_PositionableResamplingAudioSource.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/audio_sources/juce_ResamplingAudioSource.cpp b/juce/src/juce_appframework/audio/audio_sources/juce_ResamplingAudioSource.cpp
index 30424b8..947cacf 100755
--- a/juce/src/juce_appframework/audio/audio_sources/juce_ResamplingAudioSource.cpp
+++ b/juce/src/juce_appframework/audio/audio_sources/juce_ResamplingAudioSource.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/audio_sources/juce_SequenceAudioSource.cpp b/juce/src/juce_appframework/audio/audio_sources/juce_SequenceAudioSource.cpp
index 1a4de91..f59a238 100755
--- a/juce/src/juce_appframework/audio/audio_sources/juce_SequenceAudioSource.cpp
+++ b/juce/src/juce_appframework/audio/audio_sources/juce_SequenceAudioSource.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/devices/juce_AudioDeviceManager.cpp b/juce/src/juce_appframework/audio/devices/juce_AudioDeviceManager.cpp
index 1015c93..ccc3f83 100644
--- a/juce/src/juce_appframework/audio/devices/juce_AudioDeviceManager.cpp
+++ b/juce/src/juce_appframework/audio/devices/juce_AudioDeviceManager.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/dsp/juce_AudioSampleBuffer.cpp b/juce/src/juce_appframework/audio/dsp/juce_AudioSampleBuffer.cpp
index 8bc05d8..46745f5 100755
--- a/juce/src/juce_appframework/audio/dsp/juce_AudioSampleBuffer.cpp
+++ b/juce/src/juce_appframework/audio/dsp/juce_AudioSampleBuffer.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/dsp/juce_IIRFilter.cpp b/juce/src/juce_appframework/audio/dsp/juce_IIRFilter.cpp
index dc4440c..8e7f87f 100755
--- a/juce/src/juce_appframework/audio/dsp/juce_IIRFilter.cpp
+++ b/juce/src/juce_appframework/audio/dsp/juce_IIRFilter.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/midi/juce_MidiBuffer.cpp b/juce/src/juce_appframework/audio/midi/juce_MidiBuffer.cpp
index fad79b3..eb122ec 100755
--- a/juce/src/juce_appframework/audio/midi/juce_MidiBuffer.cpp
+++ b/juce/src/juce_appframework/audio/midi/juce_MidiBuffer.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/midi/juce_MidiKeyboardState.cpp b/juce/src/juce_appframework/audio/midi/juce_MidiKeyboardState.cpp
index c7132d9..a74de5a 100755
--- a/juce/src/juce_appframework/audio/midi/juce_MidiKeyboardState.cpp
+++ b/juce/src/juce_appframework/audio/midi/juce_MidiKeyboardState.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/midi/juce_MidiMessage.cpp b/juce/src/juce_appframework/audio/midi/juce_MidiMessage.cpp
index 9cef129..cdb631c 100755
--- a/juce/src/juce_appframework/audio/midi/juce_MidiMessage.cpp
+++ b/juce/src/juce_appframework/audio/midi/juce_MidiMessage.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/midi/juce_MidiMessageSequence.cpp b/juce/src/juce_appframework/audio/midi/juce_MidiMessageSequence.cpp
index 42506d2..e8c769e 100755
--- a/juce/src/juce_appframework/audio/midi/juce_MidiMessageSequence.cpp
+++ b/juce/src/juce_appframework/audio/midi/juce_MidiMessageSequence.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/plugins/juce_KnownPluginList.cpp b/juce/src/juce_appframework/audio/plugins/juce_KnownPluginList.cpp
index 8b3308c..846a356 100755
--- a/juce/src/juce_appframework/audio/plugins/juce_KnownPluginList.cpp
+++ b/juce/src/juce_appframework/audio/plugins/juce_KnownPluginList.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/plugins/juce_PluginDirectoryScanner.cpp b/juce/src/juce_appframework/audio/plugins/juce_PluginDirectoryScanner.cpp
index 476d5cf..5a11ed7 100755
--- a/juce/src/juce_appframework/audio/plugins/juce_PluginDirectoryScanner.cpp
+++ b/juce/src/juce_appframework/audio/plugins/juce_PluginDirectoryScanner.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/plugins/juce_PluginListComponent.cpp b/juce/src/juce_appframework/audio/plugins/juce_PluginListComponent.cpp
index 24e2c42..a17d6a3 100755
--- a/juce/src/juce_appframework/audio/plugins/juce_PluginListComponent.cpp
+++ b/juce/src/juce_appframework/audio/plugins/juce_PluginListComponent.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/processors/juce_AudioProcessor.cpp b/juce/src/juce_appframework/audio/processors/juce_AudioProcessor.cpp
index ea934af..b4abcad 100644
--- a/juce/src/juce_appframework/audio/processors/juce_AudioProcessor.cpp
+++ b/juce/src/juce_appframework/audio/processors/juce_AudioProcessor.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/processors/juce_AudioProcessorGraph.cpp b/juce/src/juce_appframework/audio/processors/juce_AudioProcessorGraph.cpp
index dc90726..81f4e4c 100644
--- a/juce/src/juce_appframework/audio/processors/juce_AudioProcessorGraph.cpp
+++ b/juce/src/juce_appframework/audio/processors/juce_AudioProcessorGraph.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/processors/juce_AudioProcessorPlayer.cpp b/juce/src/juce_appframework/audio/processors/juce_AudioProcessorPlayer.cpp
index 5db0cdb..14c9034 100644
--- a/juce/src/juce_appframework/audio/processors/juce_AudioProcessorPlayer.cpp
+++ b/juce/src/juce_appframework/audio/processors/juce_AudioProcessorPlayer.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/audio/synthesisers/juce_Synthesiser.cpp b/juce/src/juce_appframework/audio/synthesisers/juce_Synthesiser.cpp
index 07908f1..3b72d49 100644
--- a/juce/src/juce_appframework/audio/synthesisers/juce_Synthesiser.cpp
+++ b/juce/src/juce_appframework/audio/synthesisers/juce_Synthesiser.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/documents/juce_UndoManager.cpp b/juce/src/juce_appframework/documents/juce_UndoManager.cpp
index a332881..4275f6a 100755
--- a/juce/src/juce_appframework/documents/juce_UndoManager.cpp
+++ b/juce/src/juce_appframework/documents/juce_UndoManager.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/events/juce_ActionListenerList.cpp b/juce/src/juce_appframework/events/juce_ActionListenerList.cpp
index 088c757..33f7b9e 100755
--- a/juce/src/juce_appframework/events/juce_ActionListenerList.cpp
+++ b/juce/src/juce_appframework/events/juce_ActionListenerList.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/events/juce_ChangeListenerList.cpp b/juce/src/juce_appframework/events/juce_ChangeListenerList.cpp
index fee5ecf..02fd84f 100755
--- a/juce/src/juce_appframework/events/juce_ChangeListenerList.cpp
+++ b/juce/src/juce_appframework/events/juce_ChangeListenerList.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/events/juce_MessageListener.cpp b/juce/src/juce_appframework/events/juce_MessageListener.cpp
index 442928d..fe30b3d 100755
--- a/juce/src/juce_appframework/events/juce_MessageListener.cpp
+++ b/juce/src/juce_appframework/events/juce_MessageListener.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/buttons/juce_Button.cpp b/juce/src/juce_appframework/gui/components/buttons/juce_Button.cpp
index ee3ee15..e0e0370 100755
--- a/juce/src/juce_appframework/gui/components/buttons/juce_Button.cpp
+++ b/juce/src/juce_appframework/gui/components/buttons/juce_Button.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/controls/juce_ComboBox.cpp b/juce/src/juce_appframework/gui/components/controls/juce_ComboBox.cpp
index cd5f547..13c19e3 100755
--- a/juce/src/juce_appframework/gui/components/controls/juce_ComboBox.cpp
+++ b/juce/src/juce_appframework/gui/components/controls/juce_ComboBox.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/controls/juce_Label.cpp b/juce/src/juce_appframework/gui/components/controls/juce_Label.cpp
index eabde79..7e8cdd4 100755
--- a/juce/src/juce_appframework/gui/components/controls/juce_Label.cpp
+++ b/juce/src/juce_appframework/gui/components/controls/juce_Label.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/controls/juce_ListBox.cpp b/juce/src/juce_appframework/gui/components/controls/juce_ListBox.cpp
index ef66f99..eb78873 100644
--- a/juce/src/juce_appframework/gui/components/controls/juce_ListBox.cpp
+++ b/juce/src/juce_appframework/gui/components/controls/juce_ListBox.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/controls/juce_Slider.cpp b/juce/src/juce_appframework/gui/components/controls/juce_Slider.cpp
index deb43c1..39660d9 100644
--- a/juce/src/juce_appframework/gui/components/controls/juce_Slider.cpp
+++ b/juce/src/juce_appframework/gui/components/controls/juce_Slider.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/controls/juce_TableHeaderComponent.cpp b/juce/src/juce_appframework/gui/components/controls/juce_TableHeaderComponent.cpp
index 5091369..b265c8c 100755
--- a/juce/src/juce_appframework/gui/components/controls/juce_TableHeaderComponent.cpp
+++ b/juce/src/juce_appframework/gui/components/controls/juce_TableHeaderComponent.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/controls/juce_TableListBox.cpp b/juce/src/juce_appframework/gui/components/controls/juce_TableListBox.cpp
index 7446280..c33e2c3 100755
--- a/juce/src/juce_appframework/gui/components/controls/juce_TableListBox.cpp
+++ b/juce/src/juce_appframework/gui/components/controls/juce_TableListBox.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/controls/juce_TextEditor.cpp b/juce/src/juce_appframework/gui/components/controls/juce_TextEditor.cpp
index 6f733f1..217cc84 100644
--- a/juce/src/juce_appframework/gui/components/controls/juce_TextEditor.cpp
+++ b/juce/src/juce_appframework/gui/components/controls/juce_TextEditor.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/controls/juce_Toolbar.cpp b/juce/src/juce_appframework/gui/components/controls/juce_Toolbar.cpp
index c99d322..f769019 100755
--- a/juce/src/juce_appframework/gui/components/controls/juce_Toolbar.cpp
+++ b/juce/src/juce_appframework/gui/components/controls/juce_Toolbar.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/controls/juce_TreeView.cpp b/juce/src/juce_appframework/gui/components/controls/juce_TreeView.cpp
index 3f7ec97..9a05172 100644
--- a/juce/src/juce_appframework/gui/components/controls/juce_TreeView.cpp
+++ b/juce/src/juce_appframework/gui/components/controls/juce_TreeView.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/filebrowser/juce_DirectoryContentsDisplayComponent.cpp b/juce/src/juce_appframework/gui/components/filebrowser/juce_DirectoryContentsDisplayComponent.cpp
index debdc9a..67aa827 100755
--- a/juce/src/juce_appframework/gui/components/filebrowser/juce_DirectoryContentsDisplayComponent.cpp
+++ b/juce/src/juce_appframework/gui/components/filebrowser/juce_DirectoryContentsDisplayComponent.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/filebrowser/juce_DirectoryContentsList.cpp b/juce/src/juce_appframework/gui/components/filebrowser/juce_DirectoryContentsList.cpp
index 7349019..b3a213e 100755
--- a/juce/src/juce_appframework/gui/components/filebrowser/juce_DirectoryContentsList.cpp
+++ b/juce/src/juce_appframework/gui/components/filebrowser/juce_DirectoryContentsList.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/filebrowser/juce_FileBrowserComponent.cpp b/juce/src/juce_appframework/gui/components/filebrowser/juce_FileBrowserComponent.cpp
index a302da3..a513ed2 100755
--- a/juce/src/juce_appframework/gui/components/filebrowser/juce_FileBrowserComponent.cpp
+++ b/juce/src/juce_appframework/gui/components/filebrowser/juce_FileBrowserComponent.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/filebrowser/juce_FilenameComponent.cpp b/juce/src/juce_appframework/gui/components/filebrowser/juce_FilenameComponent.cpp
index 93bbf08..76cc46f 100755
--- a/juce/src/juce_appframework/gui/components/filebrowser/juce_FilenameComponent.cpp
+++ b/juce/src/juce_appframework/gui/components/filebrowser/juce_FilenameComponent.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/juce_Component.cpp b/juce/src/juce_appframework/gui/components/juce_Component.cpp
index 7a15357..f74cdd5 100755
--- a/juce/src/juce_appframework/gui/components/juce_Component.cpp
+++ b/juce/src/juce_appframework/gui/components/juce_Component.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/juce_Desktop.cpp b/juce/src/juce_appframework/gui/components/juce_Desktop.cpp
index 7ad1758..4f726be 100644
--- a/juce/src/juce_appframework/gui/components/juce_Desktop.cpp
+++ b/juce/src/juce_appframework/gui/components/juce_Desktop.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/keyboard/juce_KeyMappingEditorComponent.cpp b/juce/src/juce_appframework/gui/components/keyboard/juce_KeyMappingEditorComponent.cpp
index 18a0a02..5a3a3a4 100755
--- a/juce/src/juce_appframework/gui/components/keyboard/juce_KeyMappingEditorComponent.cpp
+++ b/juce/src/juce_appframework/gui/components/keyboard/juce_KeyMappingEditorComponent.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/keyboard/juce_KeyPressMappingSet.cpp b/juce/src/juce_appframework/gui/components/keyboard/juce_KeyPressMappingSet.cpp
index e72e8f6..69cae6f 100755
--- a/juce/src/juce_appframework/gui/components/keyboard/juce_KeyPressMappingSet.cpp
+++ b/juce/src/juce_appframework/gui/components/keyboard/juce_KeyPressMappingSet.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/keyboard/juce_KeyboardFocusTraverser.cpp b/juce/src/juce_appframework/gui/components/keyboard/juce_KeyboardFocusTraverser.cpp
index bbf0866..7cfc4ca 100755
--- a/juce/src/juce_appframework/gui/components/keyboard/juce_KeyboardFocusTraverser.cpp
+++ b/juce/src/juce_appframework/gui/components/keyboard/juce_KeyboardFocusTraverser.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/layout/juce_ComponentAnimator.cpp b/juce/src/juce_appframework/gui/components/layout/juce_ComponentAnimator.cpp
index 2268ae3..126e7b6 100755
--- a/juce/src/juce_appframework/gui/components/layout/juce_ComponentAnimator.cpp
+++ b/juce/src/juce_appframework/gui/components/layout/juce_ComponentAnimator.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/layout/juce_MultiDocumentPanel.cpp b/juce/src/juce_appframework/gui/components/layout/juce_MultiDocumentPanel.cpp
index 8a5b62b..4a759c0 100755
--- a/juce/src/juce_appframework/gui/components/layout/juce_MultiDocumentPanel.cpp
+++ b/juce/src/juce_appframework/gui/components/layout/juce_MultiDocumentPanel.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/layout/juce_ScrollBar.cpp b/juce/src/juce_appframework/gui/components/layout/juce_ScrollBar.cpp
index 2bbf7b2..e4cb310 100644
--- a/juce/src/juce_appframework/gui/components/layout/juce_ScrollBar.cpp
+++ b/juce/src/juce_appframework/gui/components/layout/juce_ScrollBar.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/layout/juce_StretchableLayoutManager.cpp b/juce/src/juce_appframework/gui/components/layout/juce_StretchableLayoutManager.cpp
index 4945486..9ec43fe 100755
--- a/juce/src/juce_appframework/gui/components/layout/juce_StretchableLayoutManager.cpp
+++ b/juce/src/juce_appframework/gui/components/layout/juce_StretchableLayoutManager.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/layout/juce_TabbedButtonBar.cpp b/juce/src/juce_appframework/gui/components/layout/juce_TabbedButtonBar.cpp
index 25cecc0..b26b0b6 100644
--- a/juce/src/juce_appframework/gui/components/layout/juce_TabbedButtonBar.cpp
+++ b/juce/src/juce_appframework/gui/components/layout/juce_TabbedButtonBar.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/layout/juce_TabbedComponent.cpp b/juce/src/juce_appframework/gui/components/layout/juce_TabbedComponent.cpp
index 2575805..9ef1f5c 100755
--- a/juce/src/juce_appframework/gui/components/layout/juce_TabbedComponent.cpp
+++ b/juce/src/juce_appframework/gui/components/layout/juce_TabbedComponent.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/menus/juce_MenuBarModel.cpp b/juce/src/juce_appframework/gui/components/menus/juce_MenuBarModel.cpp
index 1d8fdc2..9eac36d 100755
--- a/juce/src/juce_appframework/gui/components/menus/juce_MenuBarModel.cpp
+++ b/juce/src/juce_appframework/gui/components/menus/juce_MenuBarModel.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/menus/juce_PopupMenu.cpp b/juce/src/juce_appframework/gui/components/menus/juce_PopupMenu.cpp
index 18073b6..d055a34 100644
--- a/juce/src/juce_appframework/gui/components/menus/juce_PopupMenu.cpp
+++ b/juce/src/juce_appframework/gui/components/menus/juce_PopupMenu.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/mouse/juce_MouseCursor.cpp b/juce/src/juce_appframework/gui/components/mouse/juce_MouseCursor.cpp
index 87860b2..3e8b27c 100755
--- a/juce/src/juce_appframework/gui/components/mouse/juce_MouseCursor.cpp
+++ b/juce/src/juce_appframework/gui/components/mouse/juce_MouseCursor.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/special/juce_ColourSelector.cpp b/juce/src/juce_appframework/gui/components/special/juce_ColourSelector.cpp
index d5e7e5f..42e0e46 100755
--- a/juce/src/juce_appframework/gui/components/special/juce_ColourSelector.cpp
+++ b/juce/src/juce_appframework/gui/components/special/juce_ColourSelector.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/special/juce_MidiKeyboardComponent.cpp b/juce/src/juce_appframework/gui/components/special/juce_MidiKeyboardComponent.cpp
index aa01571..1981287 100644
--- a/juce/src/juce_appframework/gui/components/special/juce_MidiKeyboardComponent.cpp
+++ b/juce/src/juce_appframework/gui/components/special/juce_MidiKeyboardComponent.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/special/juce_OpenGLComponent.cpp b/juce/src/juce_appframework/gui/components/special/juce_OpenGLComponent.cpp
index 43d4c0e..3715bb1 100755
--- a/juce/src/juce_appframework/gui/components/special/juce_OpenGLComponent.cpp
+++ b/juce/src/juce_appframework/gui/components/special/juce_OpenGLComponent.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 #if JUCE_OPENGL
diff --git a/juce/src/juce_appframework/gui/components/windows/juce_AlertWindow.cpp b/juce/src/juce_appframework/gui/components/windows/juce_AlertWindow.cpp
index 6933718..32fecc9 100755
--- a/juce/src/juce_appframework/gui/components/windows/juce_AlertWindow.cpp
+++ b/juce/src/juce_appframework/gui/components/windows/juce_AlertWindow.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/windows/juce_ComponentPeer.cpp b/juce/src/juce_appframework/gui/components/windows/juce_ComponentPeer.cpp
index 5fb7fe3..9572c16 100755
--- a/juce/src/juce_appframework/gui/components/windows/juce_ComponentPeer.cpp
+++ b/juce/src/juce_appframework/gui/components/windows/juce_ComponentPeer.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/windows/juce_DocumentWindow.cpp b/juce/src/juce_appframework/gui/components/windows/juce_DocumentWindow.cpp
index b649b9f..d739a30 100755
--- a/juce/src/juce_appframework/gui/components/windows/juce_DocumentWindow.cpp
+++ b/juce/src/juce_appframework/gui/components/windows/juce_DocumentWindow.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/components/windows/juce_TopLevelWindow.cpp b/juce/src/juce_appframework/gui/components/windows/juce_TopLevelWindow.cpp
index 11d3f9c..41ef6f5 100644
--- a/juce/src/juce_appframework/gui/components/windows/juce_TopLevelWindow.cpp
+++ b/juce/src/juce_appframework/gui/components/windows/juce_TopLevelWindow.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/graphics/brushes/juce_GradientBrush.cpp b/juce/src/juce_appframework/gui/graphics/brushes/juce_GradientBrush.cpp
index 6b37f4f..33e3734 100755
--- a/juce/src/juce_appframework/gui/graphics/brushes/juce_GradientBrush.cpp
+++ b/juce/src/juce_appframework/gui/graphics/brushes/juce_GradientBrush.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/graphics/colour/juce_ColourGradient.cpp b/juce/src/juce_appframework/gui/graphics/colour/juce_ColourGradient.cpp
index bada781..2cd3997 100755
--- a/juce/src/juce_appframework/gui/graphics/colour/juce_ColourGradient.cpp
+++ b/juce/src/juce_appframework/gui/graphics/colour/juce_ColourGradient.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/graphics/contexts/juce_EdgeTable.cpp b/juce/src/juce_appframework/gui/graphics/contexts/juce_EdgeTable.cpp
index 1bdf5d4..97622fa 100755
--- a/juce/src/juce_appframework/gui/graphics/contexts/juce_EdgeTable.cpp
+++ b/juce/src/juce_appframework/gui/graphics/contexts/juce_EdgeTable.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/graphics/contexts/juce_Graphics.cpp b/juce/src/juce_appframework/gui/graphics/contexts/juce_Graphics.cpp
index 0f85665..7c700f8 100755
--- a/juce/src/juce_appframework/gui/graphics/contexts/juce_Graphics.cpp
+++ b/juce/src/juce_appframework/gui/graphics/contexts/juce_Graphics.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/graphics/contexts/juce_LowLevelGraphicsPostScriptRenderer.cpp b/juce/src/juce_appframework/gui/graphics/contexts/juce_LowLevelGraphicsPostScriptRenderer.cpp
index 59df0f1..9749fb5 100755
--- a/juce/src/juce_appframework/gui/graphics/contexts/juce_LowLevelGraphicsPostScriptRenderer.cpp
+++ b/juce/src/juce_appframework/gui/graphics/contexts/juce_LowLevelGraphicsPostScriptRenderer.cpp
@@ -29,6 +29,8 @@
   ==============================================================================
 */
 
+#include <cstdio>
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/graphics/contexts/juce_LowLevelGraphicsSoftwareRenderer.cpp b/juce/src/juce_appframework/gui/graphics/contexts/juce_LowLevelGraphicsSoftwareRenderer.cpp
index 80f5122..1afb675 100755
--- a/juce/src/juce_appframework/gui/graphics/contexts/juce_LowLevelGraphicsSoftwareRenderer.cpp
+++ b/juce/src/juce_appframework/gui/graphics/contexts/juce_LowLevelGraphicsSoftwareRenderer.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/graphics/drawables/juce_DrawableComposite.cpp b/juce/src/juce_appframework/gui/graphics/drawables/juce_DrawableComposite.cpp
index 6ce78dd..72217a9 100755
--- a/juce/src/juce_appframework/gui/graphics/drawables/juce_DrawableComposite.cpp
+++ b/juce/src/juce_appframework/gui/graphics/drawables/juce_DrawableComposite.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/graphics/drawables/juce_SVGParser.cpp b/juce/src/juce_appframework/gui/graphics/drawables/juce_SVGParser.cpp
index 065b9cb..7cf9238 100755
--- a/juce/src/juce_appframework/gui/graphics/drawables/juce_SVGParser.cpp
+++ b/juce/src/juce_appframework/gui/graphics/drawables/juce_SVGParser.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/graphics/fonts/juce_GlyphArrangement.cpp b/juce/src/juce_appframework/gui/graphics/fonts/juce_GlyphArrangement.cpp
index 1020bdb..5aeed20 100755
--- a/juce/src/juce_appframework/gui/graphics/fonts/juce_GlyphArrangement.cpp
+++ b/juce/src/juce_appframework/gui/graphics/fonts/juce_GlyphArrangement.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/graphics/fonts/juce_Typeface.cpp b/juce/src/juce_appframework/gui/graphics/fonts/juce_Typeface.cpp
index a223271..c141253 100644
--- a/juce/src/juce_appframework/gui/graphics/fonts/juce_Typeface.cpp
+++ b/juce/src/juce_appframework/gui/graphics/fonts/juce_Typeface.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/graphics/geometry/juce_Path.cpp b/juce/src/juce_appframework/gui/graphics/geometry/juce_Path.cpp
index e4272d7..b4c3ea5 100755
--- a/juce/src/juce_appframework/gui/graphics/geometry/juce_Path.cpp
+++ b/juce/src/juce_appframework/gui/graphics/geometry/juce_Path.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/graphics/geometry/juce_RectangleList.cpp b/juce/src/juce_appframework/gui/graphics/geometry/juce_RectangleList.cpp
index 8928ab5..a07bf83 100755
--- a/juce/src/juce_appframework/gui/graphics/geometry/juce_RectangleList.cpp
+++ b/juce/src/juce_appframework/gui/graphics/geometry/juce_RectangleList.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/graphics/imaging/image_file_formats/juce_GIFLoader.cpp b/juce/src/juce_appframework/gui/graphics/imaging/image_file_formats/juce_GIFLoader.cpp
index 07199b2..a7e5666 100755
--- a/juce/src/juce_appframework/gui/graphics/imaging/image_file_formats/juce_GIFLoader.cpp
+++ b/juce/src/juce_appframework/gui/graphics/imaging/image_file_formats/juce_GIFLoader.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/graphics/imaging/image_file_formats/juce_JPEGLoader.cpp b/juce/src/juce_appframework/gui/graphics/imaging/image_file_formats/juce_JPEGLoader.cpp
index 47216a5..3440715 100755
--- a/juce/src/juce_appframework/gui/graphics/imaging/image_file_formats/juce_JPEGLoader.cpp
+++ b/juce/src/juce_appframework/gui/graphics/imaging/image_file_formats/juce_JPEGLoader.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../../juce_core/basics/juce_StandardHeader.h"
 
 #if JUCE_MSVC
diff --git a/juce/src/juce_appframework/gui/graphics/imaging/juce_Image.cpp b/juce/src/juce_appframework/gui/graphics/imaging/juce_Image.cpp
index 482b66c..c511aea 100755
--- a/juce/src/juce_appframework/gui/graphics/imaging/juce_Image.cpp
+++ b/juce/src/juce_appframework/gui/graphics/imaging/juce_Image.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_appframework/gui/graphics/imaging/juce_ImageCache.cpp b/juce/src/juce_appframework/gui/graphics/imaging/juce_ImageCache.cpp
index c8dc558..7f3ecea 100755
--- a/juce/src/juce_appframework/gui/graphics/imaging/juce_ImageCache.cpp
+++ b/juce/src/juce_appframework/gui/graphics/imaging/juce_ImageCache.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../../../juce_core/basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_core/containers/juce_BitArray.cpp b/juce/src/juce_core/containers/juce_BitArray.cpp
index a6abb50..eb5863f 100755
--- a/juce/src/juce_core/containers/juce_BitArray.cpp
+++ b/juce/src/juce_core/containers/juce_BitArray.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_core/containers/juce_MemoryBlock.cpp b/juce/src/juce_core/containers/juce_MemoryBlock.cpp
index 0ac0c3b..b60c1b8 100755
--- a/juce/src/juce_core/containers/juce_MemoryBlock.cpp
+++ b/juce/src/juce_core/containers/juce_MemoryBlock.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_core/cryptography/juce_BlowFish.cpp b/juce/src/juce_core/cryptography/juce_BlowFish.cpp
index b0a56eb..aa88329 100755
--- a/juce/src/juce_core/cryptography/juce_BlowFish.cpp
+++ b/juce/src/juce_core/cryptography/juce_BlowFish.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_core/cryptography/juce_MD5.cpp b/juce/src/juce_core/cryptography/juce_MD5.cpp
index a6dbc2a..9e6fdd5 100755
--- a/juce/src/juce_core/cryptography/juce_MD5.cpp
+++ b/juce/src/juce_core/cryptography/juce_MD5.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_core/io/files/juce_FileOutputStream.cpp b/juce/src/juce_core/io/files/juce_FileOutputStream.cpp
index 453a386..1a799a3 100755
--- a/juce/src/juce_core/io/files/juce_FileOutputStream.cpp
+++ b/juce/src/juce_core/io/files/juce_FileOutputStream.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_core/io/juce_OutputStream.cpp b/juce/src/juce_core/io/juce_OutputStream.cpp
index edd2fd3..55ec6b6 100644
--- a/juce/src/juce_core/io/juce_OutputStream.cpp
+++ b/juce/src/juce_core/io/juce_OutputStream.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_core/io/network/juce_Socket.cpp b/juce/src/juce_core/io/network/juce_Socket.cpp
index 5b36d82..1936d3c 100644
--- a/juce/src/juce_core/io/network/juce_Socket.cpp
+++ b/juce/src/juce_core/io/network/juce_Socket.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #ifdef _WIN32
   #include "../../../../build/win32/platform_specific_code/win32_headers.h"
   #include <winsock2.h>
diff --git a/juce/src/juce_core/io/streams/juce_BufferedInputStream.cpp b/juce/src/juce_core/io/streams/juce_BufferedInputStream.cpp
index b4c506c..ba3c86c 100644
--- a/juce/src/juce_core/io/streams/juce_BufferedInputStream.cpp
+++ b/juce/src/juce_core/io/streams/juce_BufferedInputStream.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_core/io/streams/juce_MemoryInputStream.cpp b/juce/src/juce_core/io/streams/juce_MemoryInputStream.cpp
index 9d41c44..6125dc0 100755
--- a/juce/src/juce_core/io/streams/juce_MemoryInputStream.cpp
+++ b/juce/src/juce_core/io/streams/juce_MemoryInputStream.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../../basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_core/misc/juce_Uuid.cpp b/juce/src/juce_core/misc/juce_Uuid.cpp
index 898dc2b..39be3e2 100755
--- a/juce/src/juce_core/misc/juce_Uuid.cpp
+++ b/juce/src/juce_core/misc/juce_Uuid.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_core/misc/juce_ZipFile.cpp b/juce/src/juce_core/misc/juce_ZipFile.cpp
index 2fc0e47..c6c101c 100644
--- a/juce/src/juce_core/misc/juce_ZipFile.cpp
+++ b/juce/src/juce_core/misc/juce_ZipFile.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_core/text/juce_CharacterFunctions.cpp b/juce/src/juce_core/text/juce_CharacterFunctions.cpp
index 145344d..58cc8b4 100755
--- a/juce/src/juce_core/text/juce_CharacterFunctions.cpp
+++ b/juce/src/juce_core/text/juce_CharacterFunctions.cpp
@@ -29,6 +29,8 @@
   ==============================================================================
 */
 
+#include <cstdio>
+#include <cstring>
 #ifdef _MSC_VER
   #pragma warning (disable: 4514 4996)
   #pragma warning (push)
diff --git a/juce/src/juce_core/text/juce_String.cpp b/juce/src/juce_core/text/juce_String.cpp
index a25065b..4625900 100644
--- a/juce/src/juce_core/text/juce_String.cpp
+++ b/juce/src/juce_core/text/juce_String.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #ifdef _MSC_VER
   #pragma warning (disable: 4514)
   #pragma warning (push)
diff --git a/juce/src/juce_core/text/juce_StringArray.cpp b/juce/src/juce_core/text/juce_StringArray.cpp
index d26f046..c93fe35 100755
--- a/juce/src/juce_core/text/juce_StringArray.cpp
+++ b/juce/src/juce_core/text/juce_StringArray.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_core/threads/juce_Thread.cpp b/juce/src/juce_core/threads/juce_Thread.cpp
index f4234ec..6307e25 100755
--- a/juce/src/juce_core/threads/juce_Thread.cpp
+++ b/juce/src/juce_core/threads/juce_Thread.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_core/threads/juce_ThreadPool.cpp b/juce/src/juce_core/threads/juce_ThreadPool.cpp
index bfbb145..0265829 100755
--- a/juce/src/juce_core/threads/juce_ThreadPool.cpp
+++ b/juce/src/juce_core/threads/juce_ThreadPool.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/juce/src/juce_core/threads/juce_TimeSliceThread.cpp b/juce/src/juce_core/threads/juce_TimeSliceThread.cpp
index 18a6120..22a82cd 100755
--- a/juce/src/juce_core/threads/juce_TimeSliceThread.cpp
+++ b/juce/src/juce_core/threads/juce_TimeSliceThread.cpp
@@ -29,6 +29,7 @@
   ==============================================================================
 */
 
+#include <cstring>
 #include "../basics/juce_StandardHeader.h"
 
 BEGIN_JUCE_NAMESPACE
diff --git a/jucetice/src/dependancies/libsocket/UDPSocket.cpp b/jucetice/src/dependancies/libsocket/UDPSocket.cpp
index b83db2a..a2fac67 100755
--- a/jucetice/src/dependancies/libsocket/UDPSocket.cpp
+++ b/jucetice/src/dependancies/libsocket/UDPSocket.cpp
@@ -18,6 +18,9 @@
 //    Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA.
 //    ----------------------------------------------------------------------------
 
+#include <cstring>
+#include <cstdlib>
+#include <algorithm>
 #include "UDPSocket.h"
 #ifdef WIN32
   #include <winsock.h>
diff --git a/jucetice/src/dependancies/rubberband/HighFrequencyAudioCurve.cpp b/jucetice/src/dependancies/rubberband/HighFrequencyAudioCurve.cpp
index 1bc4399..908dfab 100755
--- a/jucetice/src/dependancies/rubberband/HighFrequencyAudioCurve.cpp
+++ b/jucetice/src/dependancies/rubberband/HighFrequencyAudioCurve.cpp
@@ -12,6 +12,8 @@
     COPYING included with this distribution for more information.
 */
 
+#include <cstdlib>
+#include <algorithm>
 #include "HighFrequencyAudioCurve.h"
 
 namespace RubberBand
diff --git a/jucetice/src/dependancies/rubberband/RubberBandStretcher.cpp b/jucetice/src/dependancies/rubberband/RubberBandStretcher.cpp
index 9a401b4..ff5cba3 100755
--- a/jucetice/src/dependancies/rubberband/RubberBandStretcher.cpp
+++ b/jucetice/src/dependancies/rubberband/RubberBandStretcher.cpp
@@ -12,6 +12,8 @@
     COPYING included with this distribution for more information.
 */
 
+#include <cstdlib>
+#include <algorithm>
 #include "StretcherImpl.h"
 
 namespace RubberBand {
diff --git a/jucetice/src/dependancies/rubberband/SpectralDifferenceAudioCurve.cpp b/jucetice/src/dependancies/rubberband/SpectralDifferenceAudioCurve.cpp
index fe26e3e..d37bdfe 100755
--- a/jucetice/src/dependancies/rubberband/SpectralDifferenceAudioCurve.cpp
+++ b/jucetice/src/dependancies/rubberband/SpectralDifferenceAudioCurve.cpp
@@ -12,6 +12,8 @@
     COPYING included with this distribution for more information.
 */
 
+#include <cstdlib>
+#include <algorithm>
 #include "SpectralDifferenceAudioCurve.h"
 
 namespace RubberBand
diff --git a/jucetice/src/dependancies/rubberband/StretchCalculator.cpp b/jucetice/src/dependancies/rubberband/StretchCalculator.cpp
index 77d1c50..eff6a6e 100755
--- a/jucetice/src/dependancies/rubberband/StretchCalculator.cpp
+++ b/jucetice/src/dependancies/rubberband/StretchCalculator.cpp
@@ -12,6 +12,8 @@
     COPYING included with this distribution for more information.
 */
 
+#include <cstdlib>
+#include <algorithm>
 #include "StretchCalculator.h"
 
 #include <math.h>
diff --git a/jucetice/src/dependancies/rubberband/StretcherChannelData.cpp b/jucetice/src/dependancies/rubberband/StretcherChannelData.cpp
index ecbb9a6..2abcb3e 100755
--- a/jucetice/src/dependancies/rubberband/StretcherChannelData.cpp
+++ b/jucetice/src/dependancies/rubberband/StretcherChannelData.cpp
@@ -1,5 +1,7 @@
 /* -*- c-basic-offset: 4 indent-tabs-mode: nil -*-  vi:set ts=8 sts=4 sw=4: */
 
+#include <cstdlib>
+#include <algorithm>
 #include "StretcherChannelData.h"
 
 #include "Resampler.h"
diff --git a/jucetice/src/dependancies/rubberband/StretcherImpl.cpp b/jucetice/src/dependancies/rubberband/StretcherImpl.cpp
index 30bc529..7f3acc4 100755
--- a/jucetice/src/dependancies/rubberband/StretcherImpl.cpp
+++ b/jucetice/src/dependancies/rubberband/StretcherImpl.cpp
@@ -12,6 +12,9 @@
     COPYING included with this distribution for more information.
 */
 
+#include <cstring>
+#include <cstdlib>
+#include <algorithm>
 #include "StretcherImpl.h"
 #include "PercussiveAudioCurve.h"
 #include "HighFrequencyAudioCurve.h"
diff --git a/jucetice/src/dependancies/rubberband/StretcherProcess.cpp b/jucetice/src/dependancies/rubberband/StretcherProcess.cpp
index 1e8e7d1..1ddb477 100755
--- a/jucetice/src/dependancies/rubberband/StretcherProcess.cpp
+++ b/jucetice/src/dependancies/rubberband/StretcherProcess.cpp
@@ -12,6 +12,9 @@
     COPYING included with this distribution for more information.
 */
 
+#include <cstring>
+#include <cstdlib>
+#include <algorithm>
 #include "StretcherImpl.h"
 #include "PercussiveAudioCurve.h"
 #include "HighFrequencyAudioCurve.h"
diff --git a/jucetice/src/dependancies/rubberband/Thread.cpp b/jucetice/src/dependancies/rubberband/Thread.cpp
index 2b37875..6c1a789 100755
--- a/jucetice/src/dependancies/rubberband/Thread.cpp
+++ b/jucetice/src/dependancies/rubberband/Thread.cpp
@@ -12,6 +12,8 @@
     COPYING included with this distribution for more information.
 */
 
+#include <cstdlib>
+#include <algorithm>
 #include "Thread.h"
 
 #include <iostream>
diff --git a/jucetice/src/jucetice_OpenSoundController.cpp b/jucetice/src/jucetice_OpenSoundController.cpp
index 100b54a..50eaab4 100755
--- a/jucetice/src/jucetice_OpenSoundController.cpp
+++ b/jucetice/src/jucetice_OpenSoundController.cpp
@@ -30,6 +30,7 @@
  ==============================================================================
 */
 
+#include <cstring>
 #include "jucetice_OpenSoundController.h"
 #include "dependancies/libosc/OSCMessage.h"
 #include "dependancies/libosc/OSCBundle.h"
-- 
1.6.5.1

